### PR TITLE
Add error messages to conflict options

### PR DESCRIPTION
**Pull request by xazhao**
#### Why is this change necessary?
Some post launch improvements based on comments in this PR: https://github.com/aws/aws-sam-cli/pull/4270

- Add error messages to conflict options.
- Remove the duplicate load config function
- `_load_config_values()` is still needed because it loads config file from `.aws-sam/pipeline` instead of default location.
#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/pipeline/bootstrap/cli.py [hunk 1]
```diff
@@ -265,15 +266,29 @@ def do_cli(
         Stage,
     )
 
+    is_github_parameters_provided = github_org is not None and github_repo is not None
+    is_gitlab_parameters_provided = gitlab_group is not None and gitlab_project is not None
+
+    if permissions_provider == OPEN_ID_CONNECT:
+        if oidc_provider == GITHUB_ACTIONS:
+            if not (is_github_parameters_provided and not is_gitlab_parameters_provided and not bitbucket_repo_uuid):
+                raise click.UsageError("Github Actions can only accept parameters '--github-org' and '--github-repo'")
```
Comment (sriram-mv) [inline samcli/commands/pipeline/bootstrap/cli.py:275]: could we use the Mutex click option instead?

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: samcli/commands/pipeline/bootstrap/cli.py (hunk 1 above)
Comment (xazhao) [reply samcli/commands/pipeline/bootstrap/cli.py:275]: Sure! Actually I found we have a Mutex class after this comment. Will update the code with Mutex click option

File: samcli/commands/pipeline/bootstrap/cli.py [hunk 2]
```diff
@@ -133,22 +138,35 @@
     "we currently only support GitLab, GitHub, and Bitbucket",
```
Comment (sriram-mv) [inline samcli/commands/pipeline/bootstrap/cli.py:138]: Currently supported CI/CD systems are : <>, <>, <>

File: samcli/commands/pipeline/bootstrap/cli.py [hunk 3]
```diff
@@ -217,6 +235,7 @@ def cli(
     )  # pragma: no cover
 
 
+# pylint: disable=too-many-branches
```
Comment (sriram-mv) [inline samcli/commands/pipeline/bootstrap/cli.py:238]: Should we take this opportunity to organize the codebase a bit more?

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: samcli/commands/pipeline/bootstrap/cli.py (hunk 3 above)
Comment (xazhao) [reply samcli/commands/pipeline/bootstrap/cli.py:238]: Good catch! Will scan the codebase one more time

File: tests/unit/commands/pipeline/bootstrap/test_cli.py [hunk 4]
```diff
@@ -214,6 +208,28 @@ def test_bootstrapping_oidc_non_interactive_fails_if_missing_github_parameters(s
         environment_instance.print_resources_summary.assert_not_called()
         environment_instance.save_config_safe.assert_not_called()
 
+    @patch("samcli.lib.pipeline.bootstrap.stage.Stage")
+    def test_bootstrapping_oidc_fails_conflict_parameters(self, environment_mock):
```
Comment (sriram-mv) [inline tests/unit/commands/pipeline/bootstrap/test_cli.py:212]: Should we parameterize the set of resources that conflict with each other?

**Review by sriram-mv — APPROVED** [review]

**Review by hawflau — APPROVED** [review]

**Pull request merged**